draft-ietf-httpbis-digest-headers.md

```diff
@@ -77,442 +76,413 @@ normative:
 ~~~
 
 
-# Digest Algorithm Values {#algorithms}
-
-Digest-algorithm values are used to indicate a specific digest computation.
-
-~~~ abnf
-   digest-algorithm = token
-~~~
-
-All digest-algorithm token values are case-insensitive
-but lower case is preferred;
-digest-algorithm token values MUST be compared in a case-insensitive fashion.
+# Structured Digest Algorithm Registry {#algorithms}
 
-Every digest-algorithm defines its computation procedure and
-encoding output. Unless specified otherwise, comparison of
-encoded output is case-sensitive.
+The "HTTP Structured Digest Algorithm Registry", maintained by IANA at
+<https://www.iana.org/assignments/http-dig-alg/> registers algorithms for use
+with a range of HTTP fields.
 
-The "HTTP Digest Algorithm Values Registry",
-maintained by IANA at <https://www.iana.org/assignments/http-dig-alg/> registers
-digest-algorithm values.
 Registrations MUST include the following fields:
 
-Digest algorithm:
-: the token value.
-
-: The registry can be used to reserve token values.
-
-Status:
-: the status of the algorithm
-
-: Use "standard" for standardized algorithms without known problems;
-"experimental" or some other appropriate value
-- e.g., according to the type and status of the primary document
-in which the algorithm is defined;
-"insecure" when the algorithm is insecure;
-"reserved" when Digest algorithm references a reserved token value.
-
-Description:
-: the description of the digest-algorithm and its encoding
-
-Reference:
-: a set of pointers to the primary documents defining the digest-algorithm
-
-The associated encoding for new digest-algorithms MUST either
-be represented as a quoted string
-or MUST NOT include ";" or "," in the character sets used for the encoding.
+ - Algorithm: a friendly name of the algorithm
+ - Description: a short description of the algorithm
+ - Algorithm Key: the Structured Fields key value used in
+   `Representation-Digest`, `Content-Digest`, `Want-Representation-Digest` or
+    `Want-Content-Digest` field Dictionary member keys
+ - Status: the status of the algorithm.
+     Use "standard" for standardized algorithms without known problems;
+     "experimental" or some other appropriate value
+     - e.g. according to the type and status of the primary document
+     in which the algorithm is defined;
+     "insecure" when the algorithm is insecure;
+     "reserved" when the algorithm references a reserved token value
+ - Reference: a set of pointers to the primary documents defining the algorithm
+   and key
 
 Insecure digest algorithms MAY be used to preserve integrity against corruption, but MUST
 NOT be used in a potentially adversarial setting; for example, when signing digest fields' values for
 authenticity.
 
-The registry is initialized with the tokens listed below.
+The registry is initialized with the algorithms listed below.
 
   {: vspace="0"}
-  sha-512
-  : * Digest Algorithm: sha-512
+  SHA-512
```

Comment:
Did you really have to all-caps these?  sf-dictionary keys are lcalpha.

Also, sf-dictionary keys, as lcalpha, don't allow hyphens...

Reply:
Honest truth, I never liked the style of table initialization that digest used. You'll see that in the old text we have a list of "things" that then define the values for columns in https://www.iana.org/assignments/http-dig-alg/http-dig-alg.xhtml. Is the "thing" the algorithm (SHA-512) or is it the token (sha-512), it's ambiguous.

In 
What's in here is a hangover from proposal two which needed several columns to address the old and new header algo identifer. Because of the above ambiguity, I tried to change things. But it's not really needed here. I think we should just switch to a table format and it would make things clearer.

Reply:
Oh, I see what you were doing here.  I don't much mind.  I guess it's good because the change meant that I noticed the hyphen issue, which is a real problem.  The formatting I will leave to you.

Reply:
yeah I clearly messed up the algo key thing, I'll get that fixed up thank you

Reply:
Just a note, sf-dictionary keys :do: allow hyphens, but not as the first character:

```
key           = ( lcalpha / "*" )
                *( lcalpha / DIGIT / "_" / "-" / "." / "*" )
```

Reply:
hehe. So I guess we can stick with `sha-nnn` or drop the hypen. I don't really are much either way.

Reply:
Ah. Saved. I misread the text.